api/src/test/java/org/infinispan/api/SyncStructuresAPIDemo.java

```diff
@@ -0,0 +1,27 @@
+package org.infinispan.api;
+
+import org.infinispan.api.sync.SyncContainer;
+import org.infinispan.api.sync.SyncList;
+
+/**
+ * @author Katia Aresti
+ * @since 15.0
+ **/
+public class SyncStructuresAPIDemo {
+   public void demo() {
+      try (SyncContainer infinispan = Infinispan.create("file:///path/to/infinispan.xml").sync()) {
+
+         // A default cache multimap is the container we provide
+         SyncList<String> list = infinispan.sync().list("listName");
+
+         // set
+         list.offerFirst("value");
+         // ...
+
+         // additional caches that will contain structures that users may configure
+         SyncList<String> list2 = infinispan.sync().select("name").list("listName");
```

Comment:
IMO introducing `select` is more confusing than explicitly associating data structures with an underlying cache.

If we do `infinispan.sync().cache("name").list("key")`, then the semantics are that the user wants to perform a list operation on a given key. This is analogues to the Redis approach of associating data structures with a given key, albeit that we have the additional name-spacing provided by individual caches opposed to a single namespace cache in the Redis case.

I would suspect in a lot of cases the user will do something like:

```java
SyncCache cache = Infinispan.cache("name");
cache.list("key1").push(new Value());
cache.set("Key2").add(new Value());
```